presto-main/src/test/java/io/prestosql/execution/buffer/BenchmarkBlockSerde.java

```diff
@@ -187,30 +189,20 @@ public void setup(Iterator<Page> pagesIterator)
             dataSource = sliceOutput.slice();
         }
 
-        public void setup(Type type, Iterator<?> values)
+        public <T> void setup(Type type, Iterator<T> values, BiConsumer<BlockBuilder, T> writer)
                 throws Exception
         {
             pagesSerde = new TestingPagesSerdeFactory(new TestingBlockEncodingSerde(), false).createPagesSerde();
             PageBuilder pageBuilder = new PageBuilder(ImmutableList.of(type));
             BlockBuilder blockBuilder = pageBuilder.getBlockBuilder(0);
             ImmutableList.Builder<Page> pagesBuilder = ImmutableList.builder();
             while (values.hasNext()) {
-                Object value = values.next();
+                T value = values.next();
                 if (value == null) {
                     blockBuilder.appendNull();
                 }
-                else if (BIGINT.equals(type)) {
```

Comment:
Why remove these? Now every `TypeBenchmarkData` needs to know how to serialize valus.

Reply:
Seemed more object-oriented. But I don't see any significant value here so I'll revert